docs/api-communication-spec.md

```diff
@@ -0,0 +1,686 @@
+
+| Parameter        | Type       | Required? | Default | Description                                                                                                                                                         |
+|------------------|------------|-----------|-----|---------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|                  | `Common`   | yes       | | All [common fields](#common-fields-for-pcap-agent) are included in a CF capture request                                                                                                   |
+| `token`          | `string`   | yes       | | The UAA token for the user sending the capture request.                                                                                                             |
+| `deployment`     | `string`   | yes       | | The name of the target BOSH deployment.                                                                                                                             |
+| `groups`         | `[]string` | yes       |    | A list of instance groups from which to capture. **Must contain at least one instance group**.                                                                      |
+| `instance_ids`   | `[]int`    | no        | `[]` | List of instance indexes of the application. An empty list indicates that **all instances** should be captured. Mutually exclusive with `instance_guids`.           |
+| `instance_guids` | `[]string` | no        | `[]` | List of instance IDs for finer-grained targeting. An empty list indicates that **all instances** should be captured. Mutually exclusive with `instance_ids`. |
+
+#### Stop Capture Request
+
+The stop capture request just indicates that the capture on the current stream is requested to be stopped gracefully.
+
+### Message
+
+| Parameter      | Type               | Required? | Description                                                                                                                                          |
+|----------------|--------------------|-----------|------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `message_type` | `enum MessageType` | yes       | The type of message sent. This allows for logic based on the message type                                                                            |
+| `origin`       | `string`           | yes       | The sender of this message, e.g. `pcap-api-[deployment guid]`, `pcap-agent-[instance_id]`                                                            |
+| `message`      | `string`           | no        | The detailed message, human readable, explaining the reason for this message. Optional but recommended, as it could be shown on the CLI to the user. |
+
+#### Enum `MessageType`
```

Comment:
Some of these should be represented using a protocol native termination status as discussed.

Reply:
Also there is some overlap, maybe we should have a more detailed in-person discussion about the possible codes. Alternatively we just leave it as it is and adapt the list as we go

Reply:
Added gRPC termination status codes to be used by the agent (for terminated connections, not for status updates).